AsyncDisplayKit/ASNetworkImageNode.mm

```diff
@@ -121,6 +126,17 @@ - (void)setImage:(UIImage *)image
 
 - (void)_setImage:(UIImage *)image
 {
+  // when the defaultImage is stretchable, the layer's contentsGravity is set to kCAGravityResize
+  // this will ensure the contentMode is properly set when the final image is loaded
+  if (_hasStretchableDefaultImage) {
+    if (!_needsToApplyPendingContentMode && ASObjectIsEqual(image, _defaultImage)) {
+      _pendingContentModeState = self.contentMode;
+      _needsToApplyPendingContentMode = YES;
+    } else if (_needsToApplyPendingContentMode && !ASObjectIsEqual(image, _defaultImage)) {
+      _needsToApplyPendingContentMode = NO;
+      self.contentMode = _pendingContentModeState;
+    }
+  }
```

Comment:
Minor – we should guard the new flags with `__instanceLock__`. Unfortunately that will mean either:

- Holding the lock while calling into super (who will probably reacquire the lock)
- Relinquishing the lock before calling into super (and breaking the atomicity of these operations)

I don't really like either of those options, but I think option #1 is the best for the time being. When it comes to recursive locks, "in for a penny, in for a pound" 😈